contracts/Market.sol

```diff
@@ -277,32 +277,48 @@ contract Market is Ownable, ERC721 {
 		_settle(index, result);
 	}
 
-	function _settle(uint256 id, bool result) private {
+	function settleMarket(bytes16 marketId) external {
+		_settleMarketByRange(marketId, 0, _marketBets[marketId].length - 1);
+	}
+
+	function settleMarketByRange(bytes16 marketId, uint256 from, uint256 to) external {
+		_settleMarketByRange(marketId, from, to);	
+	}
+
+	function _settleMarketByRange(bytes16 marketId, uint256 from, uint256 to) private {
+		assert(from <= _marketBets[marketId].length);
+		assert(to <= _marketBets[marketId].length);
+
+		for (uint256 i = from; i <= to; i++) {
+			uint256 tokenId = _marketBets[marketId][i];
+			Bet memory bet = _bets[tokenId];
+			if (bet.settled == false) {
+				bool result = IOracle(_oracle).checkResult(
+					marketId,
+					bet.propositionId
+				);
+				_settle(tokenId, result);
+			}
+		}
+	}
+
+	function _settle(uint256 tokenId, bool result) private {
 		require(
-			_bets[id].payoutDate < block.timestamp,
+			_bets[tokenId].payoutDate < block.timestamp,
 			"_settle: Payout date not reached"
 		);
 
-        _bets[id].settled = true;
-        _totalInPlay -= _bets[id].amount;
-        _totalExposure -= _bets[id].payout - _bets[id].amount;
-        _inplayCount --;
+        _bets[tokenId].settled = true;
+        _totalInPlay -= _bets[tokenId].amount;
+        _totalExposure -= _bets[tokenId].payout - _bets[tokenId].amount;
+        _inplayCount -= 1;
 
         address underlying = _vault.asset();
+		result == true ? IERC20(underlying).transfer(_bets[tokenId].owner, _bets[tokenId].payout) : IERC20(underlying).transfer(address(_vault), _bets[tokenId].payout);
```

Comment:
why a ternary and not a traditional if statement? IMO it is more readable to check `if(result)`

Reply:
Trying to save gas, not sure if it does tho.

```
        if (result == true) {
            // Transfer the win to the punter
            IERC20(underlying).transfer(_bets[id].owner, _bets[id].payout);
        }

        if (result == false) {
            // Transfer the proceeds to the vault, less market margin
            IERC20(underlying).transfer(address(_vault), _bets[id].payout);
        }
```

Reply:
@bazmatic thoughts? ^^^

Reply:
I also prefer straight if-then blocks for readability. I can't see how it would affect gas significantly.

Are you saying that `if (result == true)` is cheaper than just `if (result)`? Surely that won't have a significant impact on gas? Not sure why you aren't using `else`. 

IMO, unless there is a big savings, paying a few extra wei to make the code more readable is preferred.